common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/ide/util/ProjectContentGenerator.xtend

```diff
@@ -21,53 +24,98 @@ import org.eclipse.core.runtime.IProgressMonitor
-			var configurationManager = MavenPlugin.getProjectConfigurationManager()
-			var configuration = new ResolverConfiguration();
-			configuration.setResolveWorkspaceProjects(true);
-			configuration.setSelectedProfiles("");
-			configurationManager.enableMavenNature(project, configuration, monitor)
-			configurationManager.updateProjectConfiguration(project, monitor)
+			initAml = project.getFile(SRC_FOLDER + "/" + project.name + "/" + project.name + ".aml")
+			setupMavenInProject(project, fixtures, monitor)
 		}
 		if (buildsystem == GRADLE) {
-			var buildFile = project.getFile("build.gradle")
-			buildFile.create(new StringInputStream(getBuildGradleContent(fixtures)), IResource.NONE, monitor)
+			initAml = project.getFile(SRC_TEST_FOLDER + "/" + project.name + "/" + project.name + ".aml")
+			setupGradleInProject(project, fixtures, monitor)
 		}
-		project.getFolder("src/main/java/" + project.name).create(true, false, new NullProgressMonitor())
-		var initAml = project.getFile("src/main/java/" + project.name + "/" + project.name + ".aml")
-		var String amlContent = null
 		if (demo) {
 			amlContent = getDemoAMLContent(fixtures, project.name)
 			for (fixture : fixtures) {
-				createDemoTestCase(fixture, project)
+				createDemoTestCase(fixture, project, SRC_TEST_FOLDER)
 			}
 		} else {
 			amlContent = getInitialAMLContent(fixtures, project.name)
 		}
-		initAml.create(new ByteArrayInputStream(amlContent.getBytes()), IResource.NONE, monitor)
+		initAml.create(new StringInputStream(amlContent), IResource.NONE, monitor)
+		if (buildsystem == GRADLE) {
+			setupEclipseMetaData(project, monitor)
+		}
 	}
 
-	protected def createDemoTestCase(String fixture, IProject project) {
+	protected def setupEclipseMetaData(IProject project, IProgressMonitor monitor) {
+		val List<String> command = new ArrayList<String>();
+		command.add(project.location.toFile.toString + File.separator + "gradlew");
+		command.add("eclipse");
+		val ProcessBuilder processBuilder = new ProcessBuilder()
+		processBuilder.inheritIO()
+		processBuilder.redirectErrorStream(true)
+		processBuilder.command(command)
+		processBuilder.directory(project.location.toFile)
+		logger.info("Create eclipse project with gradle command {}", command)
+		val process = processBuilder.start()
+		try {
+			process.waitFor();
+		} catch (InterruptedException e) {
+			logger.info("Error", e)
+		}
+		logger.debug("Project {} refreshed", project)
+		project.refreshLocal(IProject.DEPTH_INFINITE, monitor)
+	}
+
+	protected def setupGradleInProject(IProject project, String[] fixtures, IProgressMonitor monitor) {
+		var IFile buildFile = project.getFile("build.gradle")
+		buildFile.create(new StringInputStream(getBuildGradleContent(fixtures)), IResource.NONE, monitor)
+		val bundleLocation = fileLocatorService.findBundleFileLocationAsString("org.testeditor.dsl.common")
+		val src = new File(bundleLocation + File.separator + "gradlewrapper")
```

Comment:
`File` has a specific constructor for this use-case. `new File(bundleLocation, "gradlewrapper")`